logstash-core/spec/logstash/settings/writable_directory_spec.rb

```diff
@@ -1,12 +1,17 @@
 # encoding: utf-8
 require "spec_helper"
 require "logstash/settings"
-require "stud/temporary"
+require "tmpdir"
 require "socket" # for UNIXSocket
 
+
+
+
 describe LogStash::Setting::WritableDirectory do
   let(:mode_rx) { 0555 }
-  let(:parent) { Stud::Temporary.pathname }
+  # linux is 108, Macos is 104, so use a safe value
+  # Stud::Temporary.pathname, will exceed that size without adding anything
+  let(:parent) { File.join(Dir.tmpdir, Time.now.to_f.to_s) }
```

Comment:
should we perhaps fix this in stud? we could detect in the `.pathname` method that the filename is too big a retry with a shorter hex? Or even make the length of the hex an argument and lower it to 15 instead of 30?